test/tests/pbs_Rrecord_resources_used.py

```diff
@@ -0,0 +1,616 @@
+        self.assertNotEqual(m, None)
+
+    def test_t2(self):
+        """
+        Scenario: The node on which the job was running goes down and node_fail_requeue time-out is hit 
+                  and mom is restarted with '-r'
+        Expected outcome: Server should record last known resource usage in the 'R' record.
+        """
+        jid1,jid2,jid3s1 = self.common(False,'r')
+
+        m = self.server.accounting_match(
+        msg='.*R;' + jid1 + '.*resources_used.*run_count=1', id=jid1, regexp=True)
+        self.assertNotEqual(m, None)
+        m = self.server.accounting_match(
+        msg='.*R;' + jid2 + '.*resources_used.*run_count=1', id=jid2, regexp=True)
+        self.assertNotEqual(m, None)
+        # Use re.replace() as accounting_match cannot match subjobs []
+        m = self.server.accounting_match(
+        msg='.*R;' + re.escape(jid3s1) + '.*resources_used.*run_count=1', id=jid3s1, regexp=True)
+
+    def test_t3(self):
+        """
+        Scenario: One non-rerunnable job. The node on which the job was running goes down and node_fail_requeue time-out is hit. 
+        Expected outcome: Server should record last known resource usage in the 'R' record only for rerunnable jobs.
+        """
+        a = {ATTR_JobHistoryEnable: 1}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        jid1,jid2,jid3s1 = self.common(True,'r')
+
+        m = self.server.accounting_match(
+        msg='.*R;' + jid1 + '.*resources_used.*run_count=1', id=jid1, regexp=True)
+        self.assertNotEqual(m, None)
+        m = self.server.accounting_match(
+        msg='.*R;' + jid2 + '.*resources_used.*run_count=1', id=jid2, regexp=True)
+        self.assertEqual(m, None)
+        # Use re.replace() as accounting_match cannot match subjobs []
+        m = self.server.accounting_match(
+        msg='.*R;' + re.escape(jid3s1) + '.*resources_used.*run_count=1', id=jid3s1, regexp=True)
+        self.assertNotEqual(m, None)
+
+    def test_t4(self):
+        """
+        Scenario: Mom restarted without '-r' option and requeued using qrerun.
+        Expected outcome: Server should record last known resource usage in the 'R' record for both.
+        """
+        jid1,jid2,jid3s1 = self.common(False,'s')
+
+        m = self.server.accounting_match(
+        msg='.*R;' + jid1 + '.*resources_used.*run_count=1', id=jid1, regexp=True)
+        self.assertNotEqual(m, None)
+        m = self.server.accounting_match(
+        msg='.*R;' + jid2 + '.*resources_used.*run_count=1', id=jid2, regexp=True)
+        self.assertNotEqual(m, None)
+        # Use re.replace() as accounting_match cannot match subjobs []
+        m = self.server.accounting_match(
+        msg='.*R;' + re.escape(jid3s1) + '.*resources_used.*run_count=1', id=jid3s1, regexp=True)
+        self.assertNotEqual(m, None)
+        """
+        Verify that the jobs are in 'Q' state.
+        """
+        self.server.expect(JOB, {ATTR_state: 'Q'}, jid1, offset=5)
```

Comment:
Scheduling is not false, you have a race condition between your stat and the scheduler running the jobs